client/tests/MapNavigatorImpl.test.ts

```diff
@@ -0,0 +1,118 @@
+    var startNodeId = -6;
+    var mapNodeFilter = new MapNodeFilterMock(3);
+    var errorMessage = "Start location not valid.";
+
+    expect(() => mapNavigator.findPathToNearestNode(startNodeId, mapNodeFilter)).toThrowError(errorMessage);
+  });
+});
+
+describe('testing findPathToNearestNode() with HallwayNodes', () => {
+
+  var hallwaysGraph = new HallwaysGraphMock;
+  var mapNavigator = new MapNavigatorImpl(hallwaysGraph);
+
+  test('should return correct path from 0 to 8', () => {
+    var startNodeId = 0;
+    var mapNodeFilter = new MapNodeFilterMock(8);
+    var expectedPath = [new HallwayNode(0, 1, 0, 0, 0, 0), 
+                        new HallwayNode(1, 1, 0, 0, 0, 0), 
+                        new HallwayNode(2, 1, 0, 0, 0, 0), 
+                        new HallwayNode(8, 1, 0, 0, 0, 0)];
+  
+    var path = mapNavigator.findPathToNearestNode(startNodeId, mapNodeFilter)
+    expect(path).toEqual(expectedPath);
+  });
+
+  test('should return correct path from 0 to 4', () => {
+    var startNodeId = 0;
+    var mapNodeFilter = new MapNodeFilterMock(4);
+    var expectedPath = [new HallwayNode(0, 1, 0, 0, 0, 0), 
+                        new HallwayNode(7, 1, 0, 0, 0, 0), 
+                        new HallwayNode(6, 1, 0, 0, 0, 0), 
+                        new HallwayNode(5, 1, 0, 0, 0, 0),
+                        new HallwayNode(4, 1, 0, 0, 0, 0)];
+  
+    var path = mapNavigator.findPathToNearestNode(startNodeId, mapNodeFilter)
+    expect(path).toEqual(expectedPath);
+  });
+
+  test('should return correct path from 4 to 1', () => {
+    var startNodeId = 4;
+    var mapNodeFilter = new MapNodeFilterMock(1);
+    var expectedPath = [new HallwayNode(4, 1, 0, 0, 0, 0), 
+                        new HallwayNode(5, 1, 0, 0, 0, 0), 
+                        new HallwayNode(2, 1, 0, 0, 0, 0), 
+                        new HallwayNode(1, 1, 0, 0, 0, 0)];
+  
+    var path = mapNavigator.findPathToNearestNode(startNodeId, mapNodeFilter)
+    expect(path).toEqual(expectedPath);
+  });
+
+  test('should return only one node', () => {
+    var startNodeId = 8;
+    var mapNodeFilter = new MapNodeFilterMock(8);
+    var expectedPath = [new HallwayNode(8, 1, 0, 0, 0, 0)];
+  
+    var path = mapNavigator.findPathToNearestNode(startNodeId, mapNodeFilter)
+    expect(path).toEqual(expectedPath);
+  });
+
+  test('should throw destination error', () => {
+    var startNodeId = 7;
+    var mapNodeFilter = new MapNodeFilterMock(9);
+    var errorMessage = "Destination cannot be reached.";
+
+    expect(() => mapNavigator.findPathToNearestNode(startNodeId, mapNodeFilter)).toThrowError(errorMessage);
+  });
+
+  test('should throw start location error', () => {
+    var startNodeId = 10;
+    var mapNodeFilter = new MapNodeFilterMock(10);
+    var errorMessage = "Start location not valid.";
+
+    expect(() => mapNavigator.findPathToNearestNode(startNodeId, mapNodeFilter)).toThrowError(errorMessage);
+  });
```

Comment:
Feel free to answer this discussion. Why are you creating separate test cases for hallway nodes? The path-finder implementation doesn't know about hallway vs endNode. It only knows about the MapNode type. Thus, it does not make sense to test it separately.

But as I can see, the test cases for hallways are completely different than for the nodes. If that is the case you should not delete them, but just replace the hallways with EndNodes in the mock data.

Am I missing something? Why do you feel this should be tested separately? Is there some part of the implementation that could break if multiple classes are used?

Reply:
You aren't missing anything. There was no need to test the Hallway nodes separately, I just wanted to test a different graph. I will change the misleading name, but do I have to replace the mock data with EndNodes? You said it yourself it doesn't make a difference.